massa-ledger-worker/src/ledger_db.rs

```diff
@@ -333,22 +343,24 @@ impl LedgerDB {
             ))(input)?;
             *Rc::get_mut(&mut last_key).ok_or_else(|| {
                 nom::Err::Error(nom::error::Error::new(input, nom::error::ErrorKind::Fail))
-            })? = key.clone();
+            })? = Some(key.clone());
             self.put_entry_value(handle, &mut batch, &key, &value);
             Ok((rest, ()))
         })(data)
         .map_err(|_| ModelsError::SerializeError("Error in deserialization".to_string()))?;
 
-        // Every byte should have been read
-        if last_key.is_empty() {
-            Ok(StreamingStep::Finished(None))
-        } else if rest.is_empty() {
-            self.write_batch(batch);
-            Ok(StreamingStep::Ongoing((*last_key).clone()))
-        } else {
-            Err(ModelsError::SerializeError(
-                "rest is not empty.".to_string(),
-            ))
+        match last_key.as_ref() {
+            Some(last_key) => {
+                if rest.is_empty() {
+                    self.write_batch(batch);
+                    Ok(StreamingStep::Finished(Some(last_key.clone())))
```

Comment:
This doesn't match the previous behavior it should be : 
```suggestion
                    Ok(StreamingStep::Ongoing(Some(last_key.clone())))
```